processor/decoder.go

```diff
@@ -0,0 +1,36 @@
+package processor
```

Comment:
What do you think about also moving this to the `decoder` package?

Reply:
good idea, done here: 85ec52dcc2637ac4b674f325373dfdb6f7f80f2f

we can leave the map[string]interface{} to common.MapStr to another iteration